hw/drivers/sensors/tsl2591/src/tsl2591.c

```diff
@@ -0,0 +1,706 @@
+                        int_time | gain);
+    if (rc) {
+        goto err;
+    }
+
+    #if MYNEWT_VAL(TSL2591_ITIME_DELAY)
+    /* Set the intergration time delay value in ms (+8% margin of error) */
+    g_tsl2591_itime_delay_ms = (int_time + 1) * 108;
+    #endif
+
+    /* Increment the timing changed counter */
+    STATS_INC(g_tsl2591stats, timing_changed);
+
+    return 0;
+err:
+    return rc;
+}
+
+int
+tsl2591_get_integration_time(struct sensor_itf *itf, uint8_t *itime)
+{
+    int rc;
+    uint8_t reg;
+
+    rc = tsl2591_read8(itf, TSL2591_COMMAND_BIT | TSL2591_REGISTER_CONTROL,
+                       &reg);
+    if (rc) {
+        goto err;
+    }
+
+    *itime = reg & 0x07;
+
+    return 0;
+err:
+    return rc;
+}
+
+int
+tsl2591_set_gain(struct sensor_itf *itf, uint8_t gain)
+{
+    int rc;
+    uint8_t int_time;
+
+    if ((gain != TSL2591_LIGHT_GAIN_LOW)
+        && (gain != TSL2591_LIGHT_GAIN_MED)
+        && (gain != TSL2591_LIGHT_GAIN_HIGH)
+        && (gain != TSL2591_LIGHT_GAIN_MAX)) {
+            TSL2591_LOG(ERROR, "Invalid gain value\n");
+            rc = SYS_EINVAL;
+            goto err;
+    }
+
+    rc = tsl2591_get_integration_time(itf, &int_time);
+    if (rc) {
+        goto err;
+    }
+
+    rc = tsl2591_write8(itf, TSL2591_COMMAND_BIT | TSL2591_REGISTER_CONTROL,
+                        int_time | gain);
+    if (rc) {
+        goto err;
+    }
+
+    /* Increment the gain change counter */
+    STATS_INC(g_tsl2591stats, gain_changed);
+
+    return 0;
+err:
+    return rc;
+}
+
+int
+tsl2591_get_gain(struct sensor_itf *itf, uint8_t *gain)
+{
+    int rc;
+    uint8_t reg;
+
+    rc = tsl2591_read8(itf, TSL2591_COMMAND_BIT | TSL2591_REGISTER_CONTROL,
+                       &reg);
+    if (rc) {
+        goto err;
+    }
+
+    *gain = reg & 0x30;
+
+    return 0;
+err:
+    return rc;
+}
+
+int
+tsl2591_get_data_r(struct sensor_itf *itf, uint16_t *broadband, uint16_t *ir)
+{
+    int rc;
+
+    #if MYNEWT_VAL(TSL2591_ITIME_DELAY)
+    /* Insert a delay of integration time to ensure valid sample */
+    os_time_delay((OS_TICKS_PER_SEC * g_tsl2591_itime_delay_ms) / 1000);
+    #endif
+
+    /* CHAN0 must be read before CHAN1 */
+    /* See: https://forums.adafruit.com/viewtopic.php?f=19&t=124176 */
+    *broadband = *ir = 0;
+    rc = tsl2591_read16(itf, TSL2591_COMMAND_BIT |
+                        TSL2591_REGISTER_CHAN0_LOW, broadband);
+    if (rc) {
+        goto err;
+    }
+    rc = tsl2591_read16(itf, TSL2591_COMMAND_BIT |
+                        TSL2591_REGISTER_CHAN1_LOW, ir);
+    if (rc) {
+        goto err;
+    }
+
+    /* Increment the polling counter */
+    STATS_INC(g_tsl2591stats, polled);
+
+    return 0;
+err:
+    return rc;
+}
+
+int
+tsl2591_get_data(struct sensor_itf *itf, uint16_t *broadband, uint16_t *ir)
+{
+    int rc;
+    uint8_t itime;
+    uint16_t igain;
+    uint16_t maxval;
+
+    #if !(MYNEWT_VAL(TSL2591_AUTO_GAIN))
```

Comment:
Same here; #if should not be indented, and it should not cause code inside the preprocessor conditional to be indented further.